### ISSUE TITLE
TF2.0 gradient problem of using tf.nn.relu in tf.keras.Model.

### ISSUE THREAD

**Issue by LynnHo** [tensorflow/tensorflow#27455]
**System information**
- OS Platform and Distribution: Linux Ubuntu 18.04
- TensorFlow installed from: binary
- TensorFlow version: 2.0.0-alpha0
- Python version: 3.6.8

**Describe the current behavior**
I built a keras model with only a `tf.nn.relu`, but the gradient seems to be `None` after being decorated by `@tf.function`

**Code to reproduce the issue**
1. `tf.nn.relu` + `tf.keras.Model` + `@tf.function` (this is the only case that produce `None` gradient)
```python
import tensorflow as tf

z = tf.keras.Input(())
h = tf.nn.relu(z)
m = tf.keras.Model(z, h)

@tf.function
def f(x):  # with @tf.function
    with tf.GradientTape() as t:
        t.watch(x)
        z = m(x ** 2)
    return t.gradient(z, x)

print(f(tf.convert_to_tensor(10.0)))

>>> None
```

1.2 `tf.nn.relu` + `tf.keras.Model` without `@tf.function`
```python
def f(x):  # without @tf.function
    with tf.GradientTape() as t:
        t.watch(x)
        z = m(x ** 2)
    return t.gradient(z, x)

print(f(tf.convert_to_tensor(10.0)))

>>> tf.Tensor(20.0, shape=(), dtype=float32)
```

2. `tf.keras.layers.ReLU()` + `tf.keras.Model` + `@tf.function`
```python
import tensorflow as tf

z = tf.keras.Input(())
h = tf.keras.layers.ReLU()(z)
m = tf.keras.Model(z, h)

@tf.function
def f(x):  # with @tf.function
    with tf.GradientTape() as t:
        t.watch(x)
        z = m(x ** 2)
    return t.gradient(z, x)

print(f(tf.convert_to_tensor(10.0)))

>>> tf.Tensor(20.0, shape=(), dtype=float32)
```

2.2 `tf.keras.layers.ReLU()` + `tf.keras.Model` without `@tf.function`
```python
def f(x):  # without @tf.function
    with tf.GradientTape() as t:
        t.watch(x)
        z = m(x ** 2)
    return t.gradient(z, x)

print(f(tf.convert_to_tensor(10.0)))

>>> tf.Tensor(20.0, shape=(), dtype=float32)
```

3. only `tf.nn.relu`
```python
import tensorflow as tf
m = tf.nn.relu

@tf.function
def f(x):  # with @tf.function
    with tf.GradientTape() as t:
        t.watch(x)
        z = m(x ** 2)
    return t.gradient(z, x)

print(f(tf.convert_to_tensor(10.0)))

>>> tf.Tensor(20.0, shape=(), dtype=float32)
```

So, I think its the problem between `tf.nn.relu` and `tf.keras.Model`? Besides, `tf.nn.tanh` has the same problem.

**Comment by alextp** [comment]
@tomerk I think something is broken with the keras graph here since the tape isn't seeing it.

Can you take a look, or help triage to the right person?

**Comment by tomerk** [comment]
I have a fix for this that will be submitted soon.
